### DISCUSSION TITLE
Network bridge

### DISCUSSION THREAD

**Discussion by DiamondIceNS** [TeamPneumatic/pnc-repressurized#1194] [Ideas]
I often find myself building several small, independent tube networks that utilize complex logic and/or logistics to automate some self-contained task, and I want all of them to be fed air from a central compressor room, just like in a real factory. But I don't want the limited pool of 16 color channels on the feeder network to be consumed by all the signaling functions of the independently functioning subnetworks. That, and/or I don't want any of the logistics modules between the subnetwork and its feeding supernetwork to interact.

It would be nice to have a bridge component, either in the form of its own special "dumb tube" block or an inline tube module, that allows air to freely flow through it, but acts as a barrier to redstone network signals and logistics requests.

Compare Applied Energistics, another mod that has a pipe-like network that simultaneously facilitates both power and logistics. It has the glass cable block that allows adjacent networks to share a power, but keeps their logistical networks separated. This allows you to create tiny subnets that have only one responsibility, but still receive power from a larger supernet without chewing up that supernet's available network channels.

This kind of behavior can already be achieved in PneumaticCraft, but only in a way I would describe as a hack. The solution is to put a dummy block that accepts air from multiple sides, such as a liquid compressor, in between two tube networks. This _works_, but I'm not happy about the immersion break of having random compressors in my tube network purposefully not functioning as compressors.

Alternatively, a large pressure chamber with many valves can be constructed, with a separate feeder line running to all subnetworks. This is not ideal if the pressure chamber isn't centrally located, requiring many redundant tube lines to fan out to all subnetworks. It also means all outputs of such a feeder hub would be limited to a measly 5 bar.

I'm playing on 1.19. If this already exists in 1.20, well, fantastic!

**Comment by desht** [comment]
You could actually use the Tube Junction for this - its intended purpose is to allow two separate tubes to cross without connecting, but it also doesn't forward logistics connections, so would work here.  And it's basically just a neutral tube block (with a 20 bar pressure limit), so won't lead to any unwanted interactions like a compressor would.

Let me know if that works for you...

**Reply by DiamondIceNS** [reply]
That's perfect, thank you!

I just assumed the tube junction block forwarded logistics like any other tube. I haven't had a need for its primary function yet, so I have not constructed one to test with it. Thinking about it now, though, it seems obvious that it would block logistics connections for the same reason the compressor does -- it's just a neutral pressure-accepting terminus that happens to have more than one side.

It would be helpful if the Patchouli book noted this behavior. It's not what this block is _for_, per se, and it's not the only block that can do this. But I do think this block is uniquely positioned as the next best thing for the reasons you described, and I think that makes it worth mentioning.

**Reply by desht** [reply]
Yeah, I can update the docs for that. It has also occurred to me that there are times when you would want the junction to forward logistics. Could be toggled with either a GUI control of some kind, or maybe a second junction block type (obtained via a self crafting recipe maybe?)
